Sprint_6

Lizabudka opened the pull request:
В проект добавлены:
1\ файлы описания страниц (папка pages)
2\ файлы с тестами (папка test)
3\ файл с внешними зависимостями (requirements.txt) 
4\ отчет о тестировании allure (папка allure_results)

lizasoma committed:
В проект добавлены:
1\ файлы описания страниц (папка pages)
2\ файлы с тестами (папка test)
3\ файл с внешними зависимостями (requirements.txt)
4\ отчет о тестировании allure (папка allure_results)

k-asadova commented on conftest.py:
```diff
@@ -0,0 +1,6 @@
+from selenium import webdriver
+
+
+#@pytest.fixture
```
А где фикстура? :(

k-asadova commented on pages/main_page.py:
```diff
@@ -0,0 +1,77 @@
+from selenium.webdriver.support import expected_conditions
+
+class MainPageScooterService:
+    main_page_url = 'https://qa-scooter.praktikum-services.ru/'
```
Можно лучше: константы обозначаем капслоком

k-asadova commented on pages/main_page.py:
```diff
@@ -0,0 +1,77 @@
+
+class MainPageScooterService:
+    main_page_url = 'https://qa-scooter.praktikum-services.ru/'
+    logo_yandex = [By.XPATH, './/*[@alt="Yandex"]']
```
Можно лучше: локаторы в отдельный файл

k-asadova commented on pages/main_page.py:
```diff
@@ -0,0 +1,77 @@
+    def click_question_faq_section(self, question_line):
+        element = self.driver.find_element(*self.questions_locators[question_line])
+        self.driver.execute_script("arguments[0].scrollIntoView()", element)
+        time.sleep(1)
```
Нужно поправить: не используем sleep, только WebDriverWait и expected_condition

Lizabudka replied:
К сожалению, я не понимаю, как правильно прописать WebDriverWait, если пишу expected_conditions.element_to_be_clickable или visibility_of_element_located, то почти все тесты с вопросами падают с ошибкой ElementClickInterceptedException. Т.е. кликнуть по элементу не получается, но почему так, не понимаю :(
Все идеально работает только с time.sleep()

k-asadova commented on test/test_click_on_logo.py:
```diff
@@ -0,0 +1,47 @@
+        order_page.click_on_logo_samokat()
+        current_url = self.driver.current_url
+
+        assert current_url == 'https://qa-scooter.praktikum-services.ru/'
```
Можно лучше: стараемся избегать любого хардкода, выносим тексты в отдельный файл

k-asadova commented on test/test_click_on_logo.py:
```diff
@@ -0,0 +1,47 @@
+        # Клик на лого "Яндекс", проверка url
+        main_page.click_on_logo_yandex()
+        self.driver.switch_to.window(self.driver.window_handles[1])
+        time.sleep(1)
```
Нужно поправить: sleep в тестах избегаем

k-asadova commented on requirements.txt:
```diff
@@ -0,0 +1,21 @@
+iniconfig==2.0.0
+outcome==1.3.0.post0
+packaging==24.1
+pluggy==1.5.0
```
Кажется, сюда попали библиотеки, которые не используются в проекте - нужно почистить файл

k-asadova commented on test/test_complete_order.py:
```diff
@@ -0,0 +1,49 @@
+    driver = None
+
+    @classmethod
+    def setup_class(cls):
```
Это лучше делать через фикстуру

lizasoma committed:
Внесены изменения:
1\ файл с фикстурой содержит открытие и закрытие драйвера
2\ Все константы и переменные с текстом перенесены в отдельный файл
3\ Локаторы перенесены в отдельный файл
4\ Все выражения time.sleep() убраны и заменены на WebDriverWait и expected_condition
5\ Почищены библиотеки в файле requirements.txt
6\ Добавлены описания шагов для отчета в allure
